Replace Lens with Monokle and VSCode (#860)
elastisys/compliantkubernetes: docs/user-guide/kubernetes-ui.md

---
description: How to work with Monokle or VSCode and Elastisys Compliant Kubernetes, the security-focused Kubernetes distribution.
---

# Kubernetes UI

Compliant Kubernetes recommends operations to be done via script -- if possible the GitOps way -- to improve repeatability and application developer productivity, as well as reduce human errors.
That being said, User Interfaces (UIs) have their advantages.

> In particular, they have a reduces learning curve and make features more discoverable.

Reviewer: ```suggestion
In particular, they have a reduced learning curve and make features more discoverable.
```
or 
```suggestion
In particular, they reduce the learning curve and make features more discoverable.
```

    sudo ln -s ~/.krew/bin/kubectl-oidc_login /usr/local/bin
    ```

## Monokle

To get started with Monokle:

1.  Make sure you performing the [above changes](#note-for-macos-and-linux-users).
1.  Install [Monokle](https://monokle.io/) as instructed in the upstream documentation.
1.  Perform OpenID authentication via your browser.

> 1.  Make sure to select a valid namespace, since Compliant Kubernetes does not come with a `default` namespace.

Reviewer: It does come by default in Kubernetes, though we don't give access to it as a security measure.
```suggestion
1.  Make sure to select a valid namespace, since Compliant Kubernetes does not allow access to the `default` namespace.
```
